src/textual/geometry.py

```diff
@@ -137,100 +153,121 @@ class Region(NamedTuple):
+        """Construct a Region from the top-left and bottom-right corners.
 
         Args:
-            x1 (int): Top left x
-            y1 (int): Top left y
-            x2 (int): Bottom right x
-            y2 (int): Bottom right y
+            x1 (int): Top-left x
+            y1 (int): Top-left y
+            x2 (int): Bottom-right x
+            y2 (int): Bottom-right y
 
         Returns:
             Region: A new region.
         """
         return cls(x1, y1, x2 - x1, y2 - y1)
 
     @classmethod
-    def from_origin(cls, origin: tuple[int, int], size: tuple[int, int]) -> Region:
-        """Create a region from origin and size.
+    def from_origin(cls: type[R], origin: tuple[int, int], size: tuple[int, int]) -> R:
+        """Create a new region from `origin` and `size`.
 
         Args:
-            origin (Point): Origin (top left point)
-            size (tuple[int, int]): Dimensions of region.
+            origin (tuple[int, int]): The origin (top-left point) of the new region.
+            size (tuple[int, int]): The dimensions of the new region.
 
         Returns:
-            Region: A region instance.
+            Region: A new `Region` instance.
         """
         x, y = origin
         width, height = size
         return cls(x, y, width, height)
 
     def __bool__(self) -> bool:
+        """Return True if the width and height of the region are both nonzero."""
         return bool(self.width and self.height)
 
     @property
     def x_extents(self) -> tuple[int, int]:
-        """Get the starting and ending x coord.
+        """Get the starting and ending x-coordinates of the region.
 
-        The end value is non inclusive.
+        The end value is non-inclusive.
 
         Returns:
-            tuple[int, int]: [description]
+            tuple[int, int]: A tuple describing the minimum and maximum x-coordinates.
         """
-        return (self.x, self.x + self.width)
+        x = self.x
+        return (x, x + self.width)
 
     @property
     def y_extents(self) -> tuple[int, int]:
-        """Get the starting and ending x coord.
+        """Get the starting and ending y-coordinates of the region.
 
-        The end value is non inclusive.
+        The end value is non-inclusive.
 
         Returns:
-            tuple[int, int]: [description]
+            tuple[int, int]: A tuple describing the minimum and maximum y-coordinates.
         """
-        return (self.y, self.y + self.height)
+        y = self.y
+        return (y, y + self.height)
 
     @property
     def x_max(self) -> int:
+        """Get the ending (non-inclusive) x-coordinate, an integer."""
         return self.x + self.width
 
     @property
     def y_max(self) -> int:
+        """Get the ending (non-inclusive) y-coordinate, an integer."""
         return self.y + self.height
 
     @property
     def area(self) -> int:
-        """Get the area within the region."""
+        """Get the area within the region, an integer."""
```

Comment:
It's probably better to add an `Returns:` section here, and keep the initial sentence brief.